cmake/org.eclipse.cdt.cmake.core/src/org/eclipse/cdt/cmake/core/internal/CMakeBuildConfigurationProvider.java

```diff
@@ -42,48 +42,45 @@ public String getId() {
 	@Override
 	public synchronized ICBuildConfiguration getCBuildConfiguration(IBuildConfiguration config, String name)
 			throws CoreException {
-		if (config.getName().equals(IBuildConfiguration.DEFAULT_CONFIG_NAME)) {
-			IToolChain toolChain = null;
 
-			// try the toolchain for the local target
-			Map<String, String> properties = new HashMap<>();
-			properties.put(IToolChain.ATTR_OS, Platform.getOS());
-			properties.put(IToolChain.ATTR_ARCH, Platform.getOSArch());
-			IToolChainManager toolChainManager = Activator.getService(IToolChainManager.class);
-			for (IToolChain tc : toolChainManager.getToolChainsMatching(properties)) {
+		IToolChain toolChain = null;
+
+		// try the toolchain for the local target
+		Map<String, String> properties = new HashMap<>();
+		properties.put(IToolChain.ATTR_OS, Platform.getOS());
+		properties.put(IToolChain.ATTR_ARCH, Platform.getOSArch());
+		IToolChainManager toolChainManager = Activator.getService(IToolChainManager.class);
+		for (IToolChain tc : toolChainManager.getToolChainsMatching(properties)) {
+			toolChain = tc;
+			break;
+		}
+
+		// local didn't work, try and find one that does
+		if (toolChain == null) {
+			for (IToolChain tc : toolChainManager.getToolChainsMatching(new HashMap<>())) {
 				toolChain = tc;
 				break;
 			}
+		}
 
-			// local didn't work, try and find one that does
-			if (toolChain == null) {
-				for (IToolChain tc : toolChainManager.getToolChainsMatching(new HashMap<>())) {
-					toolChain = tc;
-					break;
-				}
-			}
-
-			if (toolChain != null) {
+		if (toolChain != null) {
+			if (config.getName().equals(IBuildConfiguration.DEFAULT_CONFIG_NAME)) {
 				return new CMakeBuildConfiguration(config, name, toolChain);
 			} else {
-				// No valid combinations
-				return null;
+				ICMakeToolChainFile file = manager.getToolChainFileFor(toolChain);
+				if (file == null) {
+					Collection<ICMakeToolChainFile> files = manager.getToolChainFilesMatching(properties);
+					if (!files.isEmpty()) {
+						file = files.iterator().next();
+					}
+				}
+				if (file != null) {
+					return new CMakeBuildConfiguration(config, name, toolChain, file, "debug"); //$NON-NLS-1$
+				}
```

Comment:
This is enforcing the user to have a mandatory ICMakeToolChainFile for building in "Debug" launch mode and I believe that's not mandatory.

If we takeout `if (file != null) ` condition then we could avoid the Build not configured error in the case of the Build+Debug launch mode scenario for CMake projects.

Reply:
You are right. It is OK if file == null.
I update the PR.
Thanks.